allennlp/models/bidaf.py

```diff
@@ -101,6 +102,12 @@ def __init__(self, vocab: Vocabulary,
         span_end_input_dim = encoding_dim * 4 + span_end_encoding_dim
         self._span_end_predictor = TimeDistributed(torch.nn.Linear(span_end_input_dim, 1))
         initializer(self)
+
+        if text_field_embedder.get_output_dim() != phrase_layer.get_input_dim():
+            raise ConfigurationError("The output dimension of the text_field_embedder (embedding_dim + "
```

Comment:
Codecov is saying these lines aren't actually getting hit.  Any idea why?

Also, there are many more places in BiDAF that could use similar checks.  It's annoying to do some of them, but I think they should be done.  I can do them if you want.